src/blockchain/blockchain_monitor.ts

```diff
@@ -0,0 +1,131 @@
+import { SignedTransaction, TransactionId } from "@fleet-sdk/common";
+import { Component } from "../component.ts";
+import { ErgomaticConfig } from "../config.ts";
+import { BlockchainClient } from "./clients/mod.ts";
+
+interface MonitorState {
+  currentHeight: number;
+  /** map of txid -> bool indicating if mempool tx has been passed to plugins */
+  mempoolTxDelivery: Record<TransactionId, boolean>;
+  /** map of txid -> int indicating the number of re-checks */
+  mempoolTxChecks: Record<TransactionId, number>;
+  pastMempoolTxIds: TransactionId[];
+}
+
+interface BlockchainMonitorEvent {
+  "monitor:mempool-tx": CustomEvent<SignedTransaction>;
+  "monitor:mempool-tx-drop": CustomEvent<SignedTransaction>;
+  "monitor:included-tx": CustomEvent<SignedTransaction>;
+  "monitor:new-block": CustomEvent<unknown>;
+}
+
+export class BlockchainMonitor extends Component<BlockchainMonitorEvent> {
+  readonly #blockchainClient: BlockchainClient;
+  readonly #pollInterval: number;
+  readonly #maxMempoolTxChecks: number;
+  readonly #state: MonitorState;
+  #taskHandle?: number;
+
+  constructor(
+    config: ErgomaticConfig,
+    blockchainClient: BlockchainClient,
+    pollInterval: number = 10000,
+    maxMempoolTxChecks: number = 10,
+  ) {
+    super(config, "BlockchainMonitor");
+
+    this.#pollInterval = pollInterval;
+    this.#maxMempoolTxChecks = maxMempoolTxChecks;
+    this.#blockchainClient = blockchainClient;
+    this.#state = {
+      currentHeight: 0,
+      mempoolTxDelivery: {},
+      mempoolTxChecks: {},
+      pastMempoolTxIds: [],
+    };
+  }
+
+  start(): Promise<void> {
+    // TODO: raise component:error event if monitor throws exception
+    this.#taskHandle = setInterval(() => this.#monitor(), this.#pollInterval);
+
+    return super.start();
+  }
+
+  stop(): Promise<void> {
+    clearInterval(this.#taskHandle);
+
+    return super.stop();
+  }
+
+  async #monitor() {
+    this.logger.debug("Gathering blockchain state");
+
+    const mempool = await this.#blockchainClient.getMempool();
```

Comment:
It would be good to check if the mempool has changed before to download it. The `/info/` endpoint returns a property called `lastSeenMessageTime` which returns the lest time the node received a p2p message. If I'm right, this can be used for such check.